{{cookiecutter.project_name}}/noxfile.py

```diff
@@ -1,20 +1,70 @@
 import nox
+from pathlib import Path
+import os
 
 # Options to modify nox behaviour
+nox.options.default_venv_backend = "uv|virtualenv"
 nox.options.reuse_existing_virtualenvs = True
 
+VENV_DIR = Path("./venv").resolve()
+
 @nox.session(name="docs")
-def build_docs(session):
+def build_docs(session: nox.Session) -> None:
     """Build the documentation and load it in a browser tab, rebuilding on changes."""
-    envbindir = session.bin
     session.install("-e", ".[docs]")
     with session.chdir("docs/"):
-        session.run(
-            "sphinx-autobuild",
-            "-j",
-            "auto",
-            "--open-browser",
-            "-qT",
-            ".",
-            f"{envbindir}/../tmp/html",
-        )
+        # For local development
+        if session.interactive:
+            session.run(
+                "sphinx-autobuild",
+                "-j",
+                "auto",
+                "--open-browser",
+                "-qT",
+                ".",
+                "build/html/",
+            )
+        # For CI testing if documentation builds
+        else:
+            session.run(
+                "sphinx-build",
+                "-b",
+                "html",
+                "-W",
+                "--keep-going",
+                ".",
+                "build/html/",
+            )
+
+
+@nox.session(name="generated-project-tests")
+def run_generated_project_tests(session):
+    """Run the tests for testing units inside generated project"""
+    session.install("setuptools", silent=False)
+    session.install("-e", ".[dev]", silent=False)
+    session.run("pytest", "tests/generated_project_tests")
```

Comment:
Feel free to define a function as this is repeating - 
```python
def install_and_run_tests(session, test_dir):
    """Install dependencies and run tests in the specified directory."""
    session.install("setuptools", silent=False)
    session.install("-e", ".[dev]", silent=False)
    session.run("pytest", test_dir)
```
And use it as - `install_and_run_tests(session, "tests/template_tests")` inside sessions.